.github/workflows/build.yml

```diff
@@ -0,0 +1,167 @@
+name: Build
+on:
+  push:
+    branches: [master]
+    tags: ['*']
+  pull_request:
+    types: [opened, reopened, synchronize]
+  workflow_dispatch:
+
+env:
+  cwd: ${{github.workspace}}
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}-build
+  cancel-in-progress: true
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions/cache/restore@v4
+        id: dep-cache
+        with:
+          path: ${{github.workspace}}/node_modules
+          key: npm-${{ hashFiles('package-lock.json') }}
+          
+      - name: Use Node.js 20
+        uses: actions/setup-node@v4
+        with:
+          node-version: 20
+          cache: 'npm'
+
+      - run: npm ci
+        working-directory: ${{github.workspace}}
+        
+      - name: Cache build
+        uses: actions/cache/save@v4
+        with:
+          path: ${{github.workspace}}
+          key: npm-${{ hashFiles('package-lock.json') }}
```

Comment:
I don't think this key is correct, this will restore cache (so including the build items) if the package-lock.json matches. So, if I push any change, the cache will thus start at the old version of the code? If that code version passes all tests, and if I then introduce a bug, the CI would then not flag it - right? (Since it runs tests on the old, valid, code?)

Reply:
Should be fixed by 824345f2604ec6257ecf5d2db9f389e8f93c1b1c